Helper/Order.php

```diff
@@ -1440,6 +1440,15 @@ protected function hasSamePrice($order, $transaction)
     public function deleteOrder($order)
     {
         $this->eventsForThirdPartyModules->dispatchEvent("beforeFailedPaymentOrderSave", $order);
+        $parentQuoteId = $order->getQuoteId();
```

Comment:
For legacy flow we call deleteOrderByIncrementId -> deleteOrder and dispatch event twice
For API flow we call deleteById -> deleteOrder and dispatch event once, but parentQuote is a concept that makes snse for legacy only.

Could you please refactor taking in account those two consideration.